tests/test_setup_kubernetes_job.py

```diff
@@ -175,340 +147,237 @@ def test_ensure_pod_disruption_budget_replaces_outdated():
-                        template=V1PodTemplateSpec(),
-                        replicas=3,
-                    ),
-                )
-            ),
-        )
-        mock_deployments = [
-            KubeDeployment(
-                service="kurupt",
-                instance="fm",
-                git_sha="a12345",
-                config_sha="b12345",
-                replicas=3,
-            )
-        ]
-        ret = reconcile_kubernetes_deployment(
-            kube_client=mock_kube_client,
-            service="kurupt",
-            instance="fm",
-            kube_deployments=mock_deployments,
-            soa_dir="/nail/blah",
-        )
-        assert ret == (0, None)
-        assert not mock_create_kubernetes_application.called
-        assert not mock_update_kubernetes_application.called
+        mock_deployment_wrapper.assert_called_with(mock_deploy)
 
-        # changed gitsha so update
-        mock_create_kubernetes_application.reset_mock()
-        mock_load_kubernetes_service_config_no_cache.return_value = mock.Mock(
-            get_desired_instances=mock.Mock(return_value=3),
-            get_bounce_margin_factor=mock.Mock(return_value=1),
-            format_kubernetes_app=mock.Mock(
-                return_value=V1Deployment(
-                    metadata=V1ObjectMeta(
-                        labels={
-                            "yelp.com/paasta_git_sha": "new_image",
-                            "yelp.com/paasta_config_sha": "b12345",
-                        }
-                    ),
-                    spec=V1DeploymentSpec(
-                        selector=V1LabelSelector(),
-                        template=V1PodTemplateSpec(),
-                        replicas=3,
-                    ),
-                )
-            ),
-        )
-        mock_deployments = [
-            KubeDeployment(
-                service="kurupt",
-                instance="fm",
-                git_sha="a12345",
-                config_sha="b12345",
-                replicas=3,
-            )
-        ]
-        ret = reconcile_kubernetes_deployment(
+        mock_deploy = mock.MagicMock(spec=V1StatefulSet)
+        service_config.format_kubernetes_app.return_value = mock_deploy
+        # Create StatefulSetWrapper
+        create_application_object(
             kube_client=mock_kube_client,
             service="kurupt",
             instance="fm",
-            kube_deployments=mock_deployments,
             soa_dir="/nail/blah",
         )
-        assert ret == (0, None)
-        assert not mock_create_kubernetes_application.called
-        mock_deploy = (
-            mock_load_kubernetes_service_config_no_cache.return_value.format_kubernetes_app()
-        )
-        mock_update_kubernetes_application.assert_called_with(
-            kube_client=mock_kube_client, application=mock_deploy
-        )
+        mock_stateful_set_wrapper.assert_called_with(mock_deploy)
 
-        # changed configsha so update
```

Comment:
did you capture all these cases in the new test? I'm not sure if they're 100% necessary but I'd like to keep them if we can easily

Reply:
Yes I did. Just the code is strutured differently, so I removed some that does not make sens e anymore.  But the testcases like 'changed configsha so update' are all there.